jupyter_core/command.py

```diff
@@ -116,8 +117,10 @@ def main():
     command = 'jupyter-' + subcommand
     try:
         os.execvp(command, sys.argv[1:])
-    except OSError:
-        sys.exit("jupyter: %r is not a Jupyter command" % subcommand)
+    except OSError as e:
+        if e.errno == errno.ENOENT:
+            sys.exit("jupyter: %r is not a Jupyter command" % subcommand)
+        raise
```

Comment:
It would be better than giving the wrong error message, certainly. But I'm inclined to leave the traceback in: errors other than ENOENT are not something we anticipate, so if one comes up, the user should have all the info to debug it. If we were Python 3 only, I'd just catch `FileNotFoundError` and let everything else bubble up.